vowpalwabbit/mwt.cc

```diff
@@ -24,7 +25,7 @@ struct mwt
 {
   bool namespaces[256];        // the set of namespaces to evaluate.
   v_array<policy_data> evals;  // accrued losses of features.
-  CB::cb_class* observation;
+  std::pair<bool, CB::cb_class> optional_observation;
```

Comment:
pairs default to 0 initialization, right? If not, then we should probably initialize `first` to false

Reply:
Pairs will value initialize[0] which will result in a false value for the bool[1].

- [\[0\] - https://stackoverflow.com/questions/9025792/does-the-default-constructor-of-stdpair-set-basic-types-int-etc-to-zero](https://stackoverflow.com/questions/9025792/does-the-default-constructor-of-stdpair-set-basic-types-int-etc-to-zero)
- [\[1\] - https://stackoverflow.com/questions/3803153/what-are-primitive-types-default-initialized-to-in-c](https://stackoverflow.com/questions/3803153/what-are-primitive-types-default-initialized-to-in-c)